Admin/MonitorExchangeAuthCertificate/ConfigurationAction/Build-ExchangeAuthCertificateManagementAccount.ps1

```diff
@@ -56,22 +56,20 @@ function Build-ExchangeAuthCertificateManagementAccount {
             Write-Verbose "Calling: $($MyInvocation.MyCommand)"
 
             try {
-                $roleGroupParams = @{
-                    Name             = "Auth Certificate Management"
-                    Roles            = "View-Only Configuration", "View-Only Recipients" , "Exchange Server Certificates", "Organization Client Access"
-                    Description      = "Members of this management group can create and manage Auth Certificates"
-                    DomainController = $DomainController
-                    ErrorAction      = "Stop"
-                }
                 Write-Verbose ("Trying to create 'Auth Certificate Management' role group by using Domain Controller: $($DomainController)")
-                if ($WhatIfPreference) {
-                    Write-Host ("What if: Will create role group: 'Auth Certificate Management'")
-                    $roleGroupParams.Add("WhatIf", $true)
+                if ($PSCmdlet.ShouldProcess("View-Only Configuration, View-Only Recipients, Exchange Server Certificates, Organization Client Access", "New-RoleGroup")) {
+                    $roleGroupParams = @{
+                        Name             = "Auth Certificate Management"
+                        Roles            = "View-Only Configuration", "View-Only Recipients" , "Exchange Server Certificates", "Organization Client Access"
+                        Description      = "Members of this management group can create and manage Auth Certificates"
+                        DomainController = $DomainController
+                        ErrorAction      = "Stop"
+                        WhatIf           = $WhatIfPreference
+                    }
+                    New-RoleGroup @roleGroupParams | Out-Null
                 }
 
-                New-RoleGroup @roleGroupParams | Out-Null
-
-                if (-not($WhatIfPreference)) {
+                if ($WhatIfPreference -eq $false) {
```

Comment:
Should this be part of the ShouldProcess block above?

Reply:
Hm need to check this. Will do and keep you posted here.

Reply:
Indeed, it should be part of the `ShouldProcess` block. Will update it.